mr/operator_executor.h

```diff
@@ -38,7 +38,8 @@ class OperatorExecutor : public std::enable_shared_from_this<OperatorExecutor> {
   virtual void Stop() = 0;
 
   void ExtractFreqMap(std::function<void(std::string, detail::FreqMapWrapper&&)> cb);
- protected:
+  void ExtractCounterMap(std::function<void(std::map<std::string, long>&&)> cb);
```

Comment:
You do not need std::function here.
Just use good  `std::map<std::string, long>* dest` argument.
Also, move semantics are unneeded here as well, imho. You can fill the destination without moving into it.